# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/ComputedStyle.cpp
# Commit: 2ba84268f8ac
# Full Hash: 2ba84268f8ac2baf36477e62bc03ade97966d865
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2025-09-30 17:19 +0000
# Regressor Bug: 1987945
# File Overlap Count: 1
# Overlapping Files: accessible/tests/browser/relations/browser_anchor_positioning.js
# Description:
#   Bug 1987945 - P3: Support details relation in anchored targets in remote accessibles. r=Jamie,layout-anchor-positioning-reviewers,layout-reviewers,emilio,firefox-style-system-reviewers
# ==============================================================================

diff -r 7aaaa4514127 -r 2ba84268f8ac layout/style/ComputedStyle.cpp
--- a/layout/style/ComputedStyle.cpp	Tue Sep 30 17:19:31 2025 +0000
+++ b/layout/style/ComputedStyle.cpp	Tue Sep 30 17:19:31 2025 +0000
@@ -457,4 +457,34 @@
          });
 }
 
+// XXX: This is a broad-stroke method to return true if the referenced anchors
+// in two computed style may differ. Since we do not get the actual anchor
+// names, we do not know if the difference is just a position/size/margin change
+// or if indeed the anchors changed. So we will get false positives here, hence
+// "maybe".
+bool ComputedStyle::MaybeAnchorPosReferencesDiffer(
+    const ComputedStyle* aOther) const {
+  if (!HasAnchorPosReference() || !aOther->HasAnchorPosReference()) {
+    return true;
+  }
+
+  const auto* pos = StylePosition();
+  const auto* otherPos = aOther->StylePosition();
+  if (pos->mOffset != otherPos->mOffset || pos->mWidth != otherPos->mWidth ||
+      pos->mHeight != otherPos->mHeight ||
+      pos->mMinWidth != otherPos->mMinWidth ||
+      pos->mMinHeight != otherPos->mMinHeight ||
+      pos->mMaxWidth != otherPos->mMaxWidth ||
+      pos->mMaxHeight != otherPos->mMaxHeight ||
+      pos->mPositionAnchor != otherPos->mPositionAnchor) {
+    return true;
+  }
+
+  if (StyleMargin()->mMargin != aOther->StyleMargin()->mMargin) {
+    return true;
+  }
+
+  return false;
+}
+
 }  // namespace mozilla